microhapulator/cli/pipe.py

```diff
@@ -0,0 +1,187 @@
+    return sample_names, final_file_list
+
+
+def link_or_copy_input(filenames, workdir, docopy=False):
+    seqdir = Path(workdir) / "seq"
+    seqdir.mkdir(parents=True, exist_ok=True)
+    workingfiles = list()
+    for filepath in filenames:
+        workingpath = seqdir / filepath.name
+        workingfiles.append(Path("seq") / workingpath.name)
+        if workingpath.is_file():
+            continue
+        createfunc = copy if docopy else symlink
+        createfunc(filepath.resolve(), workingpath)
+    return [str(wf) for wf in workingfiles]
+
+
+def subparser(subparsers):
+    desc = "Perform a complete end-to-end microhap analysis pipeline"
+    cli = subparsers.add_parser("pipe", description=desc)
+    cli.add_argument(
+        "-w",
+        "--workdir",
+        metavar="D",
+        default=".",
+        help="pipeline working directory; default is current directory",
+    )
+    cli.add_argument(
+        "-n",
+        "--dryrun",
+        action="store_true",
+        help="do not execute the workflow, but display what would have been done",
+    )
+    cli.add_argument(
+        "-t",
+        "--threads",
+        metavar="T",
+        type=int,
+        default=cpu_count(),
+        help="process each batch using T threads; by default, one thread per available core is used",
+    )
+    cli.add_argument(
+        "--copy-input",
+        action="store_true",
+        help="copy input files to working directory; by default, input files are symlinked",
+    )
+    cli.add_argument(
+        "--hg38",
+        default=resource_filename("microhapulator", "data/hg38.fasta.gz"),
+        help=SUPPRESS,
+        # Hidden option for testing purposes
+    )
+    cli.add_argument(
+        "markerrefr", help="path to a FASTA file containing marker reference sequences"
+    )
+    cli.add_argument("markerdefn", help="path to a TSV file containing marker definitions")
+    cli.add_argument("seqpath", help="path to a directory containing FASTQ files")
+    cli.add_argument(
+        "samples",
+        nargs="+",
+        help="list of sample names or path to .txt file containing sample names",
+    )
+
+
+def validate_panel_config(markerseqs, markerdefn):
+    defn = load_marker_definitions(markerdefn)
+    seqs = load_marker_reference_sequences(markerseqs)
+    print("[MicroHapulator] validating panel configuration", file=sys.stderr)
+    cross_check_marker_ids(
+        seqs.keys(), defn.Marker, "marker reference sequences", "marker definitions"
+    )
+
+
+def main(args):
+    validate_panel_config(args.markerrefr, args.markerdefn)
+    samples, filenames = get_input_files(args.samples, args.seqpath)
+    workingfiles = link_or_copy_input(filenames, args.workdir, docopy=args.copy_input)
+    config = dict(
+        samples=samples,
+        readfiles=workingfiles,
+        mhrefr=Path(args.markerrefr).resolve(),
+        mhdefn=Path(args.markerdefn).resolve(),
+        hg38path=args.hg38,
```

Comment:
When using a user-specified `hg38Path` the `download_full_reference` rule in the Snakefile would still execute and attempt to download the hg38 reference file. This was due to differences in relative and absolute paths. 

Can be fixed with:
```
hg38path=Path(args.hg38).resolve()
```

Reply:
Fixed in 8a01609f3b5af846d32cdd7656524cdfff4e1f99.